src/FSharpPlus/Extensions/Array.fs

```diff
@@ -128,3 +128,16 @@ module Array =
             incr i
             f !i x
         Array.choose fi a
+    
+    ///<summary>Sets the value of a index in an array</summary>
+    ///<param name="i">The index being set</param>
+    ///<param name="x">The new value</param>
+    ///<param name="a">The input array</param>
+    ///
+    ///<returns>Array with a new value at the given index</returns>
+    let setAt i x a =
+        if Array.length a > i && i >= 0 then
+            Array.set a i x
+            a
+        else
+            a
```

Comment:
I just realized this function will mutate the original Array !
Maybe you can use `Array.init` to generate a new array instead.
Now I see a bit more that value of making these functions public but we should make clear in the XML docs that they return a new collection, ie: "Returns: A new Array with a new value at the given index".